syntax = "proto3";

package front_end;

service FrontEnd {
  rpc ExecuteMentalCommand (CommandRequest) returns (StatusReply) {}
}

message CommandRequest {
    enum CommandType {
    MOVE_FORWARD = 0;
    MOVE_BACK = 1;
    MOVE_RIGHT = 2;
    MOVE_LEFT = 3;
    MOVE_UP = 4;
    MOVE_DOWN = 5;
    TURN_RIGHT = 6;
	TURN_LEFT = 7;
	RESET = 8;
  }
  CommandType command_type = 1;
}

message StatusReply {
  int32 code = 1;
}
